src/tacacs/bash/bash_tacplus/bash_tacplus.c

```diff
@@ -0,0 +1,397 @@
+/* bash_tacplus.c - Bash plugin for TACACS+ protocol.
+ * Copyright (C) 2021, Liu Hua <liuh@microsoft.com>
+ *
+ *   TACACS+ work based on pam_tacplus.c
+ *     Copyright (C) 2010, Pawel Krawczyk <pawel.krawczyk@hush.com> and
+ *     Jeroen Nijhof <jeroen@jeroennijhof.nl>
+ *
+ *   TACACS+ authorization work based on tacplus-auth.c
+ *     https://github.com/daveolson53/tacplus-auth/blob/master/tacplus-auth.c
+ *     Copyright 2016 Cumulus Networks, Inc.  All rights reserved.
+ *     Author: Dave Olson <olson@cumulusnetworks.com>
+ *     Cumulus Networks, Inc.
+ *     185 E. Dana Street
+ *     Mountain View, CA 94041
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program - see the file COPYING.
+ *
+ * See `CHANGES' file for revision history.
+ */
+ 
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <time.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <syslog.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+/* Tacacs+ lib */
+#include <libtac/libtac.h>
+
+/* Tacacs+ support lib */
+#include <libtac/support.h>
+
+/* Output syslog to mock method when build with UT */
+#if defined (BASH_PLUGIN_UT)
+#       define syslog mock_syslog
+#endif
+
+/* Config file path */
+const char *tacacs_config_file = "/etc/tacplus_servers";
+
+/* Config file attribute */
+struct stat config_file_attr;
+
+/* Tacacs server config data */
+typedef struct {
+    struct addrinfo *address;
+    const char *key;
+} tacacs_server_t;
+
+/* Tacacs control flag */
+int tacacs_ctrl;
+
+/*
+ * Output verbose log.
+ */
+void output_verbose(const char *format, ...)
+{
+  fprintf (stderr, "TACACS+: ");
```

Comment:
>  [](http://example.com/codeflow?start=9&length=1)

Use consistent format style, why there is a space here?

Reply:
It's okay if it is other's code.

Reply:
This is by design: there is no \n after we output "TACACS+: ", so this code here just output prefix, then later we will output the log content, so the final log in stderror will look like:

TACACS+: tacacs config updated.

Reply:
I am talking about the source code format. There is an extra blank between `fprintf` and `(`.

Reply:
Fixed.